cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/transaction/TransactionRunners.java

```diff
@@ -26,14 +26,34 @@
  */
 public class TransactionRunners {
 
+  // private constructor to prevent instantiation of this class.
+  private TransactionRunners() {
+  }
+
   /**
    * Executes the given {@link TxRunnable} using the given {@link TransactionRunner}.
    *
    * @param txRunner the {@link TransactionRunner} to use for txRunner execution.
-   * @param runnable the {@link TxRunnable} to be executed inside a transaction
-   * @param <X> exception type of propagate type
+   * @param runnable the {@link TxRunnable} to be executed inside a transaction.
+   * @throws RuntimeException if cause is not an instance of X. The cause is wrapped with {@link RuntimeException}
+   * if it is not already a {@link RuntimeException}.
+   */
+  public static <X extends Throwable> void run(TransactionRunner txRunner, TxRunnable runnable) {
```

Comment:
what is X for?

Reply:
In this method it is not used. I removed it. In other methods, the exception excepted to be thrown will be X. And it is passed as a parameter as Class to tell what will `TxRunnable` will throw, all the other exceptions will be thrown as RuntimeException. This is exactly same as our previous transaction system, I have opened https://issues.cask.co/browse/CDAP-14736 as an improvement